vertical-pod-autoscaler/pkg/updater/eviction/pods_eviction_restriction.go

```diff
@@ -392,3 +433,148 @@ func setUpInformer(kubeClient kube_client.Interface, kind controllerKind) (cache
 	}
 	return informer, nil
 }
+
+// CanInPlaceUpdate performs the same checks
+func (e *podsEvictionRestrictionImpl) CanInPlaceUpdate(pod *apiv1.Pod) bool {
+	if !features.Enabled(features.InPlaceOrRecreate) {
+		return false
+	}
+	cr, present := e.podToReplicaCreatorMap[GetPodID(pod)]
+	if present {
+		// If our QoS class is guaranteed, we can't change the resources without a restart
+		if pod.Status.QOSClass == apiv1.PodQOSGuaranteed {
+			return false
+		}
+
+		if IsInPlaceUpdating(pod) {
+			return false
+		}
+
+		for _, container := range pod.Spec.Containers {
+			// If some of these are populated, we know it at least understands resizing
```

Comment:
@maxcao13 have you seen this behavior? Does this field only get populated AFTER the first resize or should it be there from the start of a pod's lifetime?

Reply:
From what I've seen is that this field is always populated when a pod is created with the feature gate on.

Reply:
Seems like from here: https://github.com/kubernetes/kubernetes/blob/83bb5d570580a3f477737fec5c24ba8fc3554264/pkg/api/pod/util.go#L638

If the feature gate is not enabled, the field on created pods gets dropped. So it's probably a good indication that this is the case.